src/meta/test/balancer_validator.cpp

```diff
@@ -35,17 +35,16 @@ namespace replication {
 #endif
 #define ASSERT_FALSE(exp) dassert(!(exp), "")
 
-static void check_cure(server_load_balancer *lb,
-                       app_mapper &apps,
-                       node_mapper &nodes,
-                       ::dsn::partition_configuration &pc)
+static void check_cure(app_mapper &apps, node_mapper &nodes, ::dsn::partition_configuration &pc)
 {
+    meta_service svc;
+    partition_guardian guardian(&svc);
     pc_status ps = pc_status::invalid;
     node_state *ns;
 
     configuration_proposal_action act;
     while (ps != pc_status::healthy) {
-        ps = lb->cure({&apps, &nodes}, pc.pid, act);
+        ps = guardian.cure({&apps, &nodes}, pc.pid, act);
```

Comment:
The unit test is `guardian test`, but not the file name `balancer_validator` mean, you can also separate it from this file

Reply:
I don't think so. It`s a function used in balancer_validator, which used to test greedy load balancer